base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -1991,6 +2035,88 @@ impl fmt::Display for MetadataValue {
             MetadataValue::PrunedHeight(height) => write!(f, "Effective pruned height is {}", height),
             MetadataValue::BestBlock(hash) => write!(f, "Chain tip block hash is {}", hash.to_hex()),
             MetadataValue::HorizonData(_) => write!(f, "Horizon data"),
+            MetadataValue::DeletedBitmap(deleted) => {
+                write!(f, "Deleted Bitmap ({} indexes)", deleted.bitmap().cardinality())
+            },
+        }
+    }
+}
+
+/// A struct that wraps a LMDB transaction and provides an interface to valid operations that can be performed
+/// on the current deleted bitmap state of the blockchain.
+/// A deleted bitmap contains the MMR leaf indexes of spent UTXOs.
+struct BlockchainDeletedBitmap<'a, T> {
```

Comment:
nit: The name on this should perhaps reflect that it is meant to be used in an operation and not as a model struct that is used in other layers.

Perhaps `DeletedBitmapUpdater`, `DeletedBitmapOperation` or `DeletedBitmapEditor`, `..Mutator` idk

Reply:
The intention here (although minimally executed) is for this struct to act as a model where you both store the data and can perform operations. I was playing with this idea for how we should structure entities within the db in general. I think the cache/lazy load idea was flawed though, so I've removed that in favour of loading the model to instantiate it

Reply:
I named this `DeletedBitmapModel` but open to suggestions (including removing the whole idea and using lmdb_xxx directly as we've been doing)